src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -400,7 +410,100 @@ private IObjectCreationExpression CreateBoundObjectCreationExpressionOperation(B
             SyntaxNode syntax = boundObjectCreationExpression.Syntax;
             ITypeSymbol type = boundObjectCreationExpression.Type;
             Optional<object> constantValue = ConvertToOptional(boundObjectCreationExpression.ConstantValue);
-            return new LazyObjectCreationExpression(constructor, memberInitializers, argumentsInEvaluationOrder, isInvalid, syntax, type, constantValue);
+            return new LazyObjectCreationExpression(constructor, initializer, argumentsInEvaluationOrder, isInvalid, syntax, type, constantValue);
+        }
+
+        private IObjectOrCollectionInitializerExpression CreateBoundObjectInitializerExpressionOperation(BoundObjectInitializerExpression boundObjectInitializerExpression)
+        {
+            Lazy<ImmutableArray<IOperation>> initializers = new Lazy<ImmutableArray<IOperation>>(() => BoundObjectCreationExpression.GetChildInitializers(boundObjectInitializerExpression).SelectAsArray(n => Create(n)));
+            bool isInvalid = boundObjectInitializerExpression.HasErrors;
+            SyntaxNode syntax = boundObjectInitializerExpression.Syntax;
+            ITypeSymbol type = boundObjectInitializerExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundObjectInitializerExpression.ConstantValue);
+            return new LazyObjectOrCollectionInitializerExpression(initializers, isInvalid, syntax, type, constantValue);
+        }
+
+        private IObjectOrCollectionInitializerExpression CreateBoundCollectionInitializerExpressionOperation(BoundCollectionInitializerExpression boundCollectionInitializerExpression)
+        {
+            Lazy<ImmutableArray<IOperation>> initializers = new Lazy<ImmutableArray<IOperation>>(() => BoundObjectCreationExpression.GetChildInitializers(boundCollectionInitializerExpression).SelectAsArray(n => Create(n)));
+            bool isInvalid = boundCollectionInitializerExpression.HasErrors;
+            SyntaxNode syntax = boundCollectionInitializerExpression.Syntax;
+            ITypeSymbol type = boundCollectionInitializerExpression.Type;
+            Optional<object> constantValue = ConvertToOptional(boundCollectionInitializerExpression.ConstantValue);
+            return new LazyObjectOrCollectionInitializerExpression(initializers, isInvalid, syntax, type, constantValue);
+        }
+
+        private IMemberReferenceExpression CreateBoundObjectInitializerMemberOperation(BoundObjectInitializerMember boundObjectInitializerMember)
+        {
+            Lazy<IOperation> instance = new Lazy<IOperation>(() => new InstanceReferenceExpression(InstanceReferenceKind.Implicit, boundObjectInitializerMember.HasErrors, syntax: boundObjectInitializerMember.Syntax, type: boundObjectInitializerMember.MemberSymbol.ContainingType, constantValue: default(Optional<object>)));
```

Comment:
I will add break lines for the couple of cases here. The formatting is not consistent across this file, so I have added a comment to https://github.com/dotnet/roslyn/issues/20204#issue-235774159. Tagging @heejaechang as an FYI, we should address this in the operation node generator.